apps/api/src/domain/models/digital-text/commands/add-photograph-to-digital-text-page/photograph-added-to-digital-text-page.event-handler.spec.ts

```diff
@@ -0,0 +1,193 @@
+
+const existingView = DigitalTextViewModel.fromDigitalTextCreated(creationEvent)
+    .apply(pageAddedEvent)
+    .apply(contentAddedToPageEvent);
+
+describe(`PhotographAddedToDigitalTextPageEventHandler`, () => {
+    let testQueryRepository: IDigitalTextQueryRepository;
+
+    let photographQueryRepository: IPhotographQueryRepository;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let app: INestApplication;
+
+    let eventHandler: PhotographAddedToDigitalTextPageEventHandler;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [
+                ConfigModule.forRoot({
+                    isGlobal: true,
+                    envFilePath: buildConfigFilePath(Environment.test),
+                    cache: false,
+                }),
+                PersistenceModule.forRootAsync(),
+            ],
+            providers: [
+                {
+                    provide: DIGITAL_TEXT_QUERY_REPOSITORY_PROVIDER_TOKEN,
+                    useFactory: (connectionProvider: ArangoConnectionProvider) => {
+                        return new ArangoDigitalTextQueryRepository(connectionProvider);
+                    },
+                    inject: [ArangoConnectionProvider],
+                },
+                PhotographAddedToDigitalTextPageEventHandler,
+            ],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = app.get(ArangoDatabaseProvider);
+
+        testQueryRepository = new ArangoDigitalTextQueryRepository(connectionProvider);
+
+        photographQueryRepository = new ArangoPhotographQueryRepository(connectionProvider);
+
+        eventHandler = new PhotographAddedToDigitalTextPageEventHandler(testQueryRepository);
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.clearViews();
+
+        await testQueryRepository.create(existingView);
+
+        await photographQueryRepository.create(existingPhotographView);
+    });
+
+    describe(`when there is an existing digital text with a page added`, () => {
+        it(`should add the given photograph to the digital text page`, async () => {
+            await eventHandler.handle(photographAddedToPageEvent);
+
+            const updatedView = (await testQueryRepository.fetchById(
+                existingView.id
+            )) as DigitalTextViewModel;
+
+            const photographSearchResult = updatedView.pages.find(
```

Comment:
@bsellars and I were talking about adding a hasPage \ getPage to the view model because we keep rewriting this search everywhere.